nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js

```diff
@@ -0,0 +1,2298 @@
+        nfCommon.cleanUpTooltips(affectedProcessorContainer, 'div.referencing-component-bulletins');
+        affectedProcessorContainer.empty();
+
+        var affectedControllerServicesContainer = $('#parameter-context-affected-controller-services');
+        nfCommon.cleanUpTooltips(affectedControllerServicesContainer, 'div.referencing-component-state');
+        nfCommon.cleanUpTooltips(affectedControllerServicesContainer, 'div.referencing-component-bulletins');
+        affectedControllerServicesContainer.empty();
+
+        $('#parameter-context-affected-unauthorized-components').empty();
+        $('#parameter-referencing-components-container').empty();
+
+        // reset the last selected parameter
+        lastSelectedId = null;
+
+        // reset the current parameter context
+        currentParameterContextEntity = null;
+
+        // clean up any tooltips that may have been generated
+        nfCommon.cleanUpTooltips($('#parameter-table'), 'div.fa-question-circle');
+    };
+
+    /**
+     * Marshals the parameters in the table.
+     */
+    var marshalParameters = function () {
+        var parameters = [];
+        var table = $('#parameter-table');
+        var parameterGrid = table.data('gridInstance');
+        var parameterData = parameterGrid.getData();
+        $.each(parameterData.getItems(), function () {
+            var parameter = {
+                'name': this.name
+            };
+
+            // if the parameter has been deleted
+            if (this.hidden === true && this.previousValue !== null) {
+                // hidden parameters were removed by the user, clear the value
+                parameters.push({
+                    'parameter': parameter
+                });
+            } else if (this.isModified === true) { // the parameter is modified
+                // check if the value has changed
+                if (this.value !== this.previousValue) {
+                    parameter['sensitive'] = this.sensitive;
+
+                    // for non-sensitive values we always include the value
+                    if (!this.sensitive) {
+                        parameter['value'] = this.value;
+                    } else {
+                        // for sensitive parameters we don't know it's value so we only include the
+                        // value if it has changed or if the empty string set checkbox has been checked
+                        if (!nfCommon.isBlank(this.value) || this.isEmptyStringSet === true) {
+                            parameter['value'] = this.value;
+                        } else if (nfCommon.isBlank(this.value) && this.previousValue !== '********') {
+                            // the sensitive parameter has not ever been saved and we still need to send the value as part of the update
+                            parameter['value'] = this.previousValue;
+                        } else if (nfCommon.isNull(this.value) && this.previousValue === '********') {
```

Comment:
I think this last else if needs to go away. This will catch any edit to only a description for a sensitive parameter and set the new value to the mask `********`

Reply:
The logic for marshalling/serializing/loading of parameters has been updated in the latest commit that updates this PR.